tests/evolution/test_evolution_config_validation.py

```
@@ -0,0 +1,151 @@
+
+    def test_tournament_size_with_small_population(self):
+        """Test tournament size validation with small population."""
+        # Valid: tournament <= population
+        config = EvolutionConfig(population_size=3, tournament_size=3)
+        assert config.validate() is True
+        
+        # Invalid: tournament > population
+        config = EvolutionConfig(population_size=2, tournament_size=3)
+        assert config.validate() is False
+
+    def test_semantic_operator_config(self):
```

Ruff (0.12.2)

Missing return type annotation for public function `test_semantic_operator_config`

Add return type annotation: `None`

(ANN201)